core/io/ip_address.h

```diff
@@ -94,3 +94,7 @@ struct IPAddress {
 	IPAddress(uint32_t p_a, uint32_t p_b, uint32_t p_c, uint32_t p_d, bool is_v6 = false);
 	IPAddress() { clear(); }
 };
+
+// Zero-constructing IPAddress initializes field, valid, and wildcard to 0 (and thus empty).
+template <>
+struct is_zero_constructible<IPAddress> : std::true_type {};
```

Comment:
I don't think this is needed, `IPAddress` looks to be trivially constructible.
It might be worth adding a static assert to ensure it is (and stays) that way!

Reply:
Looking at just the variables you might think so, but it actually isn't:
```
IPAddress() { clear(); }
```
The user-defined default constructor makes it not trivially constructible. I dunno why it doesn't just setup the values in the header, but that's a cleanup for another time.